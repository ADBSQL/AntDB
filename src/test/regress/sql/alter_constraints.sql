set grammar to oracle;
--删除检查约束
create table tt(id integer unique, name char(10), constraint cons_name check( name in('a','b','c')));
alter table tt drop constraint cons_name;
drop table tt;
--删除外键约束
create table tt(id integer primary key,name char(10));
create table aa(id int primary key, job varchar(10), constraint fk FOREIGN KEY(id) REFERENCES tt(id));
alter table aa drop constraint fk;
drop table tt;
drop table aa;
--增加检查约束
create table tt(id integer unique,name char(10));
alter table tt add constraint cons_name check( name in('a','b','c'));
drop table tt;
--删除主键约束
create table tt(id integer primary key,name char(10));
alter table tt drop primary key;
drop table tt;
create table tt(id integer,name char(10),constraint pkk primary key(id));
alter table tt drop constraint pkk;
drop table tt;
