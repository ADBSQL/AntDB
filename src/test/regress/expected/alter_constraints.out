set grammar to oracle;
--删除检查约束
create table alterconsttbl(id integer unique, name char(10), constraint cons_name check( name in('a','b','c')));
alter table alterconsttbl drop constraint cons_name;
drop table alterconsttbl;
--删除外键约束
create table alterconsttbl(id integer primary key,name char(10));
create table alterconsttbl2(id int primary key, job varchar(10), constraint fk FOREIGN KEY(id) REFERENCES alterconsttbl(id));
alter table alterconsttbl2 drop constraint fk;
drop table alterconsttbl;
drop table alterconsttbl2;
--增加检查约束
create table alterconsttbl(id integer unique,name char(10));
alter table alterconsttbl add constraint cons_name check( name in('a','b','c'));
drop table alterconsttbl;
--删除主键约束
create table alterconsttbl(id integer primary key,name char(10));
alter table alterconsttbl drop primary key;
ERROR:  syntax error at or near "key"
LINE 1: alter table alterconsttbl drop primary key;
                                               ^
drop table alterconsttbl;
create table alterconsttbl(id integer,name char(10),constraint pkk primary key(id));
alter table alterconsttbl drop constraint pkk;
drop table alterconsttbl;
